pyquil/api/_quantum_computer.py

```diff
@@ -261,8 +263,34 @@ def compile(self, program: Program,
 
     def reset(self):
         """
-        Reset the QuantumComputer's QAM to its initial state.
+        Reset the QuantumComputer's QAM to its initial state, and refresh all the connection
+        objects in the event that the ~/.forest_config file has changed during the existence
+        of this QuantumComputer object.
         """
+        def refresh_client(client: Client, new_endpoint: str) -> Client:
+            timeout = client.timeout
+            client.close()
+            return Client(new_endpoint, timeout)
+
+        if isinstance(self.qam, QVM) and isinstance(self.compiler, QVMCompiler):
```

Comment:
What's your preference for doing this dispatch by hand, rather than putting these are instance methods on the corresponding *Compiler objects?

Reply:
Hmm I'm not sure I follow -- are you saying something like adding an `is_qvm()` to the `QAM` and `AbstractCompiler` base classes?

Reply:
I think @ecpeterson means having the QAM and compiler base classes having a refresh instance method. That way you don't have to manually dispatch.

Reply:
Ah I see, the "are" typo sent me in the wrong direction.

Reply:
Sure it did

Reply:
@ecpeterson @notmgsk I considered doing this stuff in the QVM/QPU and Compiler objects, but then I'd have to create the `PyquilConfig` object in like 4 different places. I think there are valid arguments for either choice -- I'm happy to go that route if you think it is better.

Reply:
Sure ya did

Reply:
You could create the object in one place and pass it around, if that bothers you? It doesn't especially bother me, though.